cni-plugin/main.go

```diff
@@ -309,6 +319,23 @@ func cmdDel(args *skel.CmdArgs) error {
 	return nil
 }
 
+func getAPIServerPorts(ctx context.Context, api *kubernetes.Clientset) ([]string, error) {
+	service, err := api.CoreV1().Services("default").Get(ctx, "kubernetes", metav1.GetOptions{})
+	if err != nil {
+		return []string{}, err
+	}
+
+	ports := make([]string, 0)
+	for _, port := range service.Spec.Ports {
```

Comment:
@stevej @mateiidavid This would be more robust if we looked at the return value of `IntValue()` rather than checking the `Type`.

For example:

```go
	for _, port := range service.Spec.Ports {
		ports = append(ports, strconv.Itoa(int(port.Port)))

		if tp := port.TargetPort.IntValue(); tp == 0 {
			logrus.Debugf("linkerd-cni: failed to convert target port, omitting: %s", port.TargetPort.String())
		} else {
			ports = append(ports, strconv.Itoa(tp))
		}
	}
```